### ISSUE TITLE
Wallabag removed while upgrading to 2.5.1~ynh2

### ISSUE THREAD

**Issue by Gofannon** [YunoHost-Apps/wallabag2_ynh#153]
### Describe the bug

1. Cannot upgrade app to latest version  (`2.3.8~ynh4` => `2.5.1~ynh2`)
2. Package cannot  restore the "pre-backup"  `wallabag2-pre-upgrade1.tar` after the failed upgrade
   1. application is removed from server
   2. restore from backup is ok

### Context

- Hardware: *dedicated computer : kimsufi box*
- YunoHost version: 11.0.9 (beta bulleyes)
- I have access to my server: *Through SSH | through the webadmin*
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: *yes*
  - If yes, please explain: upgrade to beta yunohost version since May and Wallabag is broken since
- Using, or trying to install package version/branch: stable / master
- If upgrading, current package version: *2.3.8~ynh4*

### Steps to reproduce

- from the webadmin, upgrade wallabag to the latest version (same from CLI)

### Expected behavior

1. Application should be upgraded
2. if fails, should be restore to working state as before upgrade

### Logs

```bash
2022-08-09 21:42:22,235: WARNING - 762  This backup archive can not be restored because it comes from a too-old YunoHost version.
2022-08-09 21:42:22,391: DEBUG - + [[ -d /etc/yunohost/apps/wallabag2 ]]
2022-08-09 21:42:22,392: DEBUG - + ynh_die '--message=Uhoh ... Yunohost failed to restore the app to the way it was before the failed upgrade :|'
2022-08-09 21:42:22,483: WARNING - Uhoh ... Yunohost failed to restore the app to the way it was before the failed upgrade :|
```

https://paste.yunohost.org/edoqiliwew.bash

from the log file, it seems that I am affected by https://github.com/YunoHost-Apps/wallabag2_ynh/pull/129 and needs to use the https://github.com/YunoHost-Apps/wallabag2_ynh/pull/133 to fix the database. Am I correct?

**Comment by lapineige** [comment]
I think I had this issue too, backup is not automatically restored, doing it afterwards works.

> from the log file, it seems that I am affected by #129 and needs to use the #133 to fix the database. Am I correct?

You do :)

I think the best way to fix this is to execute these commands : https://github.com/wallabag/wallabag/issues/5233#issuecomment-1041863310

I can provide assistance about how to do it (in summary : connect to your Wallabag MySQL database (password found in yunohost app settings), execute the command, run the upgrade).

**Comment by Gofannon** [comment]
> I think I had this issue too, backup is not automatically restored, doing it afterwards works.

I was afraid that other people migrating to bulleyes might get stuck with this but I was running the beta so I don't know... Time will tell with the release of YunoHost 11 :)




> You do :)
> 
> I think the best way to fix this is to execute these commands : [wallabag/wallabag#5233 (comment)](https://github.com/wallabag/wallabag/issues/5233#issuecomment-1041863310)
> 
> I can provide assistance about how to do it (in summary : connect to your Wallabag MySQL database (password found in yunohost app settings), execute the command, run the upgrade).

Well, I went the "lazy way" by using https://github.com/YunoHost-Apps/wallabag2_ynh/pull/133 as an intermediate step to fix the DB. It went smoothly

 2.3.8\~ynh4 (old)=> 2.4.3\~ynh1 (fix db) => 2.5.1\~ynh2 (current)

Thanks for all the people working of the package!

@lapineige , you can close this issue if you want, it was a false alert. 

If someone has to do the same steps:

1. fix the DB
    `yunohost app upgrade wallabag2 -u https://github.com/nicofrand/wallabag2_ynh/tree/upgrade-database-scheme`

2. then upgrade the app with normal means from YunoHost to get the last version

**Comment by lapineige** [comment]
> I was afraid that other people migrating to bulleyes might get stuck with this but I was running the beta so I don't know... Time will tell with the release of YunoHost 11 :)

Then we should know pretty soon as it was done :D

If it restores well afterwards, that's not a so big deal.

> Well, I went the "lazy way" by using #133 as an intermediate step to fix the DB. It went smoothly

Great, thank you for the feedback, this means it's working ! 🎉


> you can close this issue if you want, it was a false alert.*

It was not, this is important to document too 😉
